packages/teams-js/src/public/pages.ts

```diff
@@ -198,9 +183,9 @@ export namespace pages {
         throw errorNotSupportedOnPlatform;
       }
       if (runtime.isLegacyTeams) {
-        resolve(sendAndHandleStatusAndReason('executeDeepLink', createTeamsAppLink(params)));
+        resolve(sendAndHandleStatusAndReasonWithVersion('v2', 'executeDeepLink', createTeamsAppLink(params)));
```

Comment:
Why do we need different version numbers here? I would assume the same version numbers would be fine since there are different function names being used.

Reply:
Ideally, navigateToApp is a 'v1' API but it calls executeDeepLink internally and executeDeepLink in Teams-Js library is a 'v2' API. Let's hold this discussion and I need to ask PM about this problem. Good catch! Thank you.

Reply:
@TrevorJoelHarris, could you please explain a little bit more what happened if runtime is legacy Teams? Are we sending out different requests based on runtime? I feel like we might need a clear definition here to figure out whether it is v1, v2, or something else.